shell/platform/darwin/macos/framework/Source/FlutterCompositor.h

```diff
@@ -76,15 +78,18 @@ class FlutterCompositor {
   bool EndFrame(bool has_flutter_content);
 
   // Creates a CALayer object which is backed by the supplied IOSurface, and
-  // adds it to the root CALayer for this FlutterViewController's view.
+  // adds it to the root CALayer for the given view.
   void InsertCALayerForIOSurface(
+      FlutterView* view,
```

Comment:
What's the purpose of this change? If the compositor is inherently bound to a view provider at creation time, why do we want public API that uses an arbitrary view rather than the view provided by its provider? It seems like this makes it much easier to accidentally use the wrong view object here. Sholudn't it take a view ID instead?

Reply:
The very change of this PR is that the compositor is no longer bound to a view provider at creation. The compositor (which needs to be a singleton) needs to be able to handle multiple views that are only available after the app starts.

Reply:
Moreover, `InsertCALayerForIOSurface` knows nothing about where the view comes from. `FlutterCompositor` only knows that there's a new CALayer that needs to be removed later. The only result that an evil subclass calls this method with a strange view is that the CALayer will be removed.

Reply:
Sorry, I missed that this was protected; I thought it was fully public API.